src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/network/access/LocalPublicIpv4Provider.kt

```diff
@@ -0,0 +1,61 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.network.access
+
+import com.atlassian.performance.tools.awsinfrastructure.Ipv4Validator
+import com.atlassian.performance.tools.awsinfrastructure.VerboseUrlReader
+import java.net.URL
+import java.util.function.Supplier
+
+class LocalPublicIpv4Provider private constructor(
+    private val servicesToQuery: List<URL>,
+    private val urlReader: (serviceUrl: URL) -> String?,
+    private val ipValidator: (ip: String) -> Boolean
+) : () -> String, Supplier<String> {
+    object Defaults {
+        val servicesToQuery: List<URL> = listOf(
+            URL("https://checkip.amazonaws.com")
```

Comment:
Do you rely on the DNS lookup inside `URL.equals`? If not, we should use `URI` instead.

Reply:
I don't know about relying on it. Would you be able to explain or share link to explanation why this matters?

Reply:
[`URL.equals` Javadoc](https://docs.oracle.com/javase/9/docs/api/java/net/URL.html#equals-java.lang.Object-):
>  Two hosts are considered equivalent if both host names can be resolved into the same IP addresses; else if either host name can't be resolved, the host names must be equal without regard to case; or both host names equal to null.
> Since hosts comparison requires name resolution, this operation is a blocking operation.
> Note: The defined behavior for equals is known to be inconsistent with virtual hosting in HTTP.

Reply:
If I shouldn't use `URL` here then I'd rather use `String` for the service addresses. With URI I will have problem around spliting type conversion responsibilities.